File: src/sample/InsertionSort.java
package sample;

class InsertionSort {

    public void insertionSort(int[] array) {
        for (int i = 0; i < array.length; i++) {
            int position = i;
            int tempValue = array[position];

            while (position > 0 && array[position - 1] > tempValue) {
                array[position] = array[position - 1];
                position = position - 1;
            }
            array[position] = tempValue;
        }
    }
}

File: insertion_sort.js
function insertionSort(array) {
    var range = Array.from({length: array.length}, (x, i) => i ); 
    range.forEach((index) => {
        position = index 
        tempValue = array[index]

        while (position > 0 && array[position - 1] > tempValue) {
            array[position] = array[position -1]
            position = position - 1
        }
        array[position] = tempValue
    })
    return array; 
}

console.log(insertionSort([7, 3, 4, 3, 2, 5, 4, 1]))